src/main/kotlin/org/wycliffeassociates/resourcecontainer/DirectoryAccessor.kt

```diff
@@ -0,0 +1,25 @@
+package org.wycliffeassociates.resourcecontainer
+
+import java.io.BufferedReader
+import java.io.BufferedWriter
+import java.io.File
+
+class DirectoryAccessor(private val file: File) : IResourceContainerAccessor {
+    override fun getReader(filename: String): BufferedReader {
+        return getFile(filename).bufferedReader()
+    }
+
+    override fun checkFileExists(filename: String): Boolean {
+        return getFile(filename).exists()
+    }
+
+    private fun getFile(filename: String): File = File(file, filename)
+
+    override fun initWrite() {
+        file.mkdirs()
+    }
+
+    override fun write(filename: String, writeFcn: (BufferedWriter) -> Unit) {
```

Comment:
I think I'd prefer Fn instead of Fcn if you need to abbreviate this

Reply:
done